pushnotifications-integration-tests/src/androidTest/java/com/example/pushnotificationsintegrationtests/MultiInstanceSupportTest.kt

```diff
@@ -0,0 +1,338 @@
+package com.example.pushnotificationsintegrationtests
+
+import android.support.test.InstrumentationRegistry
+import android.support.test.runner.AndroidJUnit4
+import com.pusher.pushnotifications.BeamsCallback
+import com.pusher.pushnotifications.PushNotificationsInstance
+import com.pusher.pushnotifications.PusherCallbackError
+import com.pusher.pushnotifications.SubscriptionsChangedListener
+import com.pusher.pushnotifications.auth.TokenProvider
+import com.pusher.pushnotifications.fcm.MessagingService
+import com.pusher.pushnotifications.internal.InstanceDeviceStateStore
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.awaitility.core.ConditionTimeoutException
+import org.awaitility.kotlin.await
+import org.awaitility.kotlin.untilNotNull
+import org.awaitility.kotlin.untilNull
+import org.awaitility.kotlin.until
+import org.hamcrest.CoreMatchers.*
+import org.junit.After
+import org.junit.AfterClass
+
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import org.junit.Assert.*
+import org.junit.Before
+import java.io.File
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class MultiInstanceSupportTest {
+  val context = InstrumentationRegistry.getTargetContext()
+  val instanceId1 = "00000000-1241-08e9-b379-377c32cd1e80"
+  val instanceId2 = "00000000-1241-08e9-b379-377c32cd1e82"
+  val errolClient1 = ErrolAPI(instanceId1, "http://localhost:8080")
+  val errolClient2 = ErrolAPI(instanceId2, "http://localhost:8080")
+
+  fun getStoredDeviceId(instanceId: String): String? {
+    val deviceStateStore = InstanceDeviceStateStore(InstrumentationRegistry.getTargetContext(), instanceId)
+    return deviceStateStore.deviceId
+  }
+
+  companion object {
+    val errol = FakeErrol(8080, "really-long-cluster-secret-key")
```

Comment:
we should make "really-long-cluster-secret-key" a const, and then we can use it more safely in the tests too.